src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -329,6 +329,11 @@ def __init__(
     def vocab_size(self) -> int:
         return len(self.encoder)
 
+    @property
+    def timestamp_end(self) -> int:
+        timestamp_ids = [value for key, value in self.added_tokens_encoder.items() if self.timestamp_pat.match(key)]
+        return max(timestamp_ids)
```

Comment:
It's on this point that I'll be the most cautious, as we introduced two ways of computing `timestamp_end`:
1. In the modeling file with `timestamp_begin + generation_config.get("number_timestamp_tokens", 1501)`
2. in the tokenizer, with the current way of doing it: taking the maximum id of every tokens that match the timestamp pattern.

@itazap and @ArthurZucker, WDYT of this ?

To give a bit of context, the Whisper model has its usual vocabulary on which is appended a series of timestamps token ids (usually 1501 tokens). When a user appends a new token to the tokenizer, it is wrongly identified as a timestamp token since it's id is greater than the o.g vocabulary size.

Reply:
Would it make sense to avoid using timestamp begin and timestamp end here, and to simply verify in the rest of the file if a token is a timestamp thanks to `self.timestamp_pat.match(key)` ?

Reply:
This could be a way to be more rigorous on how we teat timestamps in the tokenizer but would introduce a difference in how we compute timestamps in the generation file and here.

Reply:
Up on this! I do agree that the introduced inconsistency is not convenient. Since I do not see another way of identifying `timestamp_end`, and that the solution I proposed is computationally unoptimal (recomputed at each call, could be improved with LRU cache), I agree that identifying timestamp tokens from the tokenizer's path would be better with a regex match. This would remove necessity of `timestamp_end`, nevertheless we can't avoid using `timestamp_begin` since it is required to computing timestamp values ([here](https://github.com/huggingface/transformers/blob/74026b473e8748706a7a86fd20d6a275306d8ffb/src/transformers/models/whisper/tokenization_whisper.py#L544)).

Reply:
overall let's aim for simplicity:
- what is the best for the users (least amount of work for them)
- what is the simplest code for this.